lib/blas_quda.cu

```diff
@@ -234,23 +234,23 @@ namespace quda {
 #if QUDA_PRECISION & 2
           if (x.Ncolor() != 3) { errorQuda("nColor = %d is not supported", x.Ncolor()); }
           if (x.Nspin() == 4 && x.FieldOrder() == QUDA_FLOAT4_FIELD_ORDER) { // wilson
-#if defined(GPU_WILSON_DIRAC) || defined(GPU_DOMAIN_WALL_DIRAC)
+#if defined(NSPIN4)
             const int M = 6;
             nativeBlas<float4, short4, short4, M, Functor, writeX, writeY, writeZ, writeW, writeV>(
                 a, b, c, x, y, z, w, v, x.Volume());
 #else
             errorQuda("blas has not been built for Nspin=%d fields", x.Nspin());
 #endif
           } else if (x.Nspin() == 4 && x.FieldOrder() == QUDA_FLOAT2_FIELD_ORDER) { // wilson
-#if defined(GPU_MULTIGRID)
+#if defined(GPU_MULTIGRID) // FIXME eventually we should get rid of this and use float4 ordering
```

Comment:
Can this be `#if defined(NSPIN2)` instead? Or is `GPU_MULTIGRID` a stronger requirement here?

Reply:
Nevermind---didn't pay close enough attention to the context.

Reply:
This is intentional (it's not `NSPIN2`), it's the funky float2 ordering that MG fields use in single precision.  And I want to remove these eventually and have MG use float4 always for `nSpin=4`.